scalafix-tests/unit/src/test/scala/scalafix/tests/DenotationOpsTest.scala

```diff
@@ -0,0 +1,32 @@
+package scalafix.tests
+
+import scala.meta._
+import scalafix.syntax._
+import scalafix.internal.util.DenotationOps
+
+class DenotationOpsTest extends BaseSemanticTest("DenotationOpsTest") {
+
+  test("resultType") {
+    val source = docs.input.parse[Source].get
+    source.collect {
+      case t @ Pat.Var(Name("x")) =>
+        for {
+          symbol <- t.symbol
+          resultType <- symbol.resultType
+        } yield assert(resultType.structure == Type.Name("Boolean").structure)
```

Comment:
is there a smarter way to compare trees?

Reply:
There is! `import scala.meta.contrib._; q"a.b".isEqual(q"a.b.")`

Reply:
thanks! I'll fix it